src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/VsItemList.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.NuGet
+{
+    /// <summary>
+    /// Abstract list with Item method for getting members by index or name
+    /// </summary>
+    internal abstract class VsItemList<T> : KeyedCollection<string, T>
+    {
+        public VsItemList() : base() { }
+
+        public VsItemList(IEnumerable<T> collection) : base()
+        {
+            Requires.NotNull(collection, nameof(collection));
+
+            foreach (var item in collection)
+            {
+                Add(item);
+            }
+        }
+
+        public T Item(Object index)
+        {
+            try 
+            {
+                if (index is string)
+                {
+                    return this[(string)index];
+                }
+                else if (index is int)
+                {
+                    return this[(int)index];
+                }
+            }
+            catch (KeyNotFoundException ex)
+            {
+                TraceUtilities.TraceException($"{index} was not found", ex);
```

Comment:
Is this invalid or not? I see you're dong this sometimes by design - if so, why trace? It just dirties debugger output.